### PR TITLE
Gen 2

### PR DESCRIPTION
**Pull request by Aegyo**
Does not yet have choosing a new starter, and no popup when you can travel to Johto. Also pokemon health on routes, dungeons, gyms are not balanced in the slightest. Shops also haven't been set up.

After all Kanto pokemon have been caught, a button appears in Pallet town which will send you to Johto. To travel back, you first have to fight your way through Johto to Olivine, where there is a Dock for travel to other regions. Similarly, there is a dock in Kanto at Vermillion City, but the player won't be able to access it until they have been to Johto once.

### CODE REVIEW

File: src/components/shipModal.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            <div class="modal-body">
+                <img src='assets/images/ship.png'>
+                <div style='margin-top:20px'>
+                    <span data-bind='ifnot: player.region == 0'>
+                        <button class='btn btn-primary' onclick='MapHelper.moveToTown("Vermillion City");player.region = 0;$("#ShipModal").modal("hide")'>
+                            Kanto
+                        </button>
+                    </span>
+                    <span data-bind='ifnot: player.region == 1'>
+                        <button class='btn btn-primary' onclick='MapHelper.moveToTown("Olivine City");player.region = 1;$("#ShipModal").modal("hide")'>
```
Comment (dennism1997) [inline src/components/shipModal.html]: Maybe it's better to move what's inside the onclick to methods in the maphelper class

Comment (Aegyo) [reply src/components/shipModal.html]: I had it like this to plan for when gen 3 gets added, and we have two buttons.

File: src/components/shipModal.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<div class="modal noselect fade" id="ShipModal" tabindex="-1" role="dialog" aria-labelledby="ShipModalLabel">
```
Comment (dennism1997) [inline src/components/shipModal.html]: What is this file?

File: src/scripts/GameConstants.ts [hunk 3]
```diff
@@ -321,7 +325,20 @@ namespace GameConstants {
...
+        "Fog" = 17,
+        "Storm" = 18,
+        "Mineral" = 19,
+        "Glacier" = 20,
+        "Rising" = 21,
+        "Will" = 22,
+        "Koga" = 23,
+        "Bruno2" = 24,
+        "Karen" = 25,
+        "JohtoChampion" = 26,
```
Comment (dennism1997) [inline src/scripts/GameConstants.ts]: Isn't Karen the Johto Champion?

File: src/components/shipModal.html (hunk 2 above)
Comment (Aegyo) [reply src/components/shipModal.html]: Modal that opens when you click one of the docks at Olivine or Vermillion. Accessed through the dungeon square next to those cities.

File: src/scripts/GameConstants.ts (hunk 3 above)
Comment (Aegyo) [reply src/scripts/GameConstants.ts]: Lance is the Johto Champion

File: src/scripts/Player.ts [hunk 4]
```diff
@@ -851,6 +864,18 @@ class Player {
         this._questPoints(value);
     }
 
+    public ableToTravel() {
```
Comment (dennism1997) [inline src/scripts/Player.ts]: These two methods may have a better place in the maphelper. At least that's where I would look for these methods

Comment (Aegyo) [reply src/scripts/Player.ts]: Yeah, I wasn't 100% sure where they should go.

File: src/scripts/worldmap/MapHelper.ts [hunk 5]
```diff
@@ -5,6 +5,7 @@ class MapHelper {
             if (this.accessToRoute(route, region)) {
                 $("[data-route='" + player.route() + "']").removeClass('currentRoute').addClass('unlockedRoute');
                 player.route(route);
+                player.region = region;
```
Comment (dennism1997) [inline src/scripts/worldmap/MapHelper.ts]: Isn't player.region an observable now? So shouldn't it be `player.region(region)`

Comment (Aegyo) [reply src/scripts/worldmap/MapHelper.ts]: player._region is observable, but we have get and set methods for player.region so this works fine

File: src/components/shipModal.html (hunk 2 above)
Comment (dennism1997) [reply src/components/shipModal.html]: Oh huh I put it at the wrong location. I meant to ask about the test.html file

File: src/scripts/GameConstants.ts (hunk 3 above)
Comment (dennism1997) [reply src/scripts/GameConstants.ts]: Oh yeah

File: src/scripts/worldmap/MapHelper.ts (hunk 5 above)
Comment (dennism1997) [reply src/scripts/worldmap/MapHelper.ts]: Cool

File: src/components/shipModal.html (hunk 2 above)
Comment (Aegyo) [reply src/components/shipModal.html]: oh, whoops, that was just so I could see the diff between the map in develop and my KantoSVG. It shouldn't have been in the pr

**Comment by BaineGames** [comment]
and so it begins

**Comment by dennism1997** [comment]
The map feels kinda cramped. Don't really know what to do about it tho. Maybe some space around the edges but not sure if that's possible

**Pull request merged**